File: src/12.ts
import { promises as fs } from 'fs'

const solve11 = async (): Promise<void> => {
    const table = (await fs.readFile('./res/popular-names.txt', 'utf-8'))
        .split('\n')
        .map(row => row.split('\t'))
    const pickColumn = (i: number): string =>
        table.map(row => row[i]).join('\n')
    fs.writeFile('./res/col1.txt', pickColumn(0))
    fs.writeFile('./res/col2.txt', pickColumn(1))
}

solve11()

File: src/12.py
# cut -f 1 res/popular-names.txt
# cut -f 2 res/popular-names.txt

from helper.file_helper import read_string, write_string

source = read_string('./res/popular-names.txt')
table = tuple(map(lambda row: tuple(row.split('\t')), source.split('\n')))
transposed = tuple(zip(*table))
write_string('./res/col1.txt', '\n'.join(transposed[0]))
write_string('./res/col2.txt', '\n'.join(transposed[1]))
